### PR TITLE
Default to sample rules and dictionary

### PR DESCRIPTION
**Pull request by localjo**
Changes the sample rules config and dictionary to be used as the defaults which are combined with custom rules and dictionaries that are passed in. Makes it easier to use the CLI for checking documentation in other projects without defining config files first.

To do:
- [x] Change file names to reflect new roles
- [x] Update index.js to use defaults when no rules are passed in
- [x] Make custom config merge with defaults when passed in
- [x] Review updates to README to make sure they are accurate
- [x] Confirm that all new features are working as intended and old features aren't broken

This is a breaking change, so it should be released with v3.0.0

### CODE REVIEW

**Comment by localjo** [comment]
- [x] Make sure arrays from the default and custom config files are properly merged instead of being overwritten.

**Comment by localjo** [comment]
This properly merges default and custom rules, so it's "working", but it introduces some new usability problems since severity levels are defined in different keys and overriding the defaults isn't always straightforward. For example, if the `filler` rule is set to be fatal in the defaults, there is no clear way to change it's severity level in the custom rules other than turning off the rule completely, which involves using a different key in the config file, so it's not intuitive what's going on. Probably the best way to handle that will be to change the config object's format to be something like;

```
{
  "dictionaries": "custom.dic",
  "rules": {
    "list-item-content-indent": false,
    "list-item-indent": {"severity": "suggest"},
    "list-item-spacing": true,
    "maximum-line-length": {"severity": "warn"},
    "no-auto-link-without-protocol": {"severity": "fatal"},
    "no-missing-blank-lines": {
      "exceptTightLists": true,
      "severity": "suggest"
    }
    "readability": {
      "age": 18,
      "minWords": 7,
      "severity": "fatal"
    }
  },
  "ignore": [
    "address",
    "attempt"
  ],
  "noIgnore": [
    "function",
    "require"
  ]
}
```

Basically, rename the `lint` key to `rules` and add an optional `severity` key to each rule's value, move the readability key into the rules key, and add a `noIgnore` key to allow overriding phrases which are ignored.

**Comment by localjo** [comment]
Added a `-v` flag which allows a user to see which rules are being applied.
